src/NServiceBus.Core/Hosting/Helpers/AssemblyScanner.cs

```diff
@@ -76,14 +74,18 @@ public AssemblyScannerResults GetScannableAssemblies()
                 {
                     if (!assembly.IsDynamic)
                     {
-                        ScanAssembly(AssemblyPath(assembly), results, processed);
+                        ScanAssembly(assembly, results);
                     }
                 }
             }
 
+            var processed = new Dictionary<string, bool>(StringComparer.OrdinalIgnoreCase);
```

Comment:
If `ScanAssembliesInAppDomain` is true we scan assemblies that are loaded into the appdomain already. But these assemblies will also lay on the disc. So in that case we might scan those assemblies again. Should we move up the processed declaration to include the assemblies scanned in the appdomain as well?